sdk/inc/azure/core/az_json.h

```diff
@@ -390,6 +390,13 @@ az_json_writer_get_bytes_used_in_destination(az_json_writer const* json_writer)
 /**
  * @brief Appends the UTF-8 text value (as a JSON string) into the buffer.
  *
+ * @note If you receive an #AZ_ERROR_NOT_ENOUGH_SPACE result while appending data for which there is
+ * theoretically space, note that the JSON writer requires at least 64-bytes of slack within the
+ * output buffer, above the theoretical minimal space needed. The JSON writer pessimistically
+ * requires at least 64-bytes of space when writing any chunk of data larger than 10 characters
+ * because it tries to write in 64 byte chunks (10 character * 6 if all need to be escaped into the
+ * unicode form).
```

Comment:
I think the 10*6 goes into too much detail about the specifics of the implementation, that might change and isn't relevant to the caller.

It also doesn't exactly apply to append APIs that aren't dealing with strings (like appending numbers).